### PR TITLE
feat: shrub search screen

### PR DESCRIPTION
**Pull request by carolynzhuang**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
* Added the shrub screens
* Added a search feature for the shrub screens
* Attempted to add navigation for the shrub screen search - unsuccessful
* I think ShrubSpeciesSearch.tsx is where the navigation issue is, but I'm not sure how to fix it (within the ToggleSwitch)
* Fixed an error with formatEnumKey where it would call .replace on a non-string
* Fixed an error with ShrubSpeciesSearch and TreeSpeciesSearch so that it is able to toggle between the 2 search screens

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
CAROLYN: Added a placeholder home screen to address some routing issues. home screen will be assigned in a later sprint

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @christophertorres1

### CODE REVIEW

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    winter: activeFilters.bloom.includes('winter'),
+    spring: activeFilters.bloom.includes('spring'),
+    summer: activeFilters.bloom.includes('summer'),
+    fall: activeFilters.bloom.includes('fall'),
+  });
+
+  const [activeSunFilters, setActiveSunFilters] = useState({
+    fullSun: activeFilters.sun_exposure.includes('full sun'),
+    partialShade: activeFilters.sun_exposure.includes('partial shade'),
+    coarseGrained: activeFilters.sun_exposure.includes('coarse-grained'),
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:59]: 'coarse-grained' should be 'shade'

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+};
+
+export const ShrubSearchFilter: React.FC<ShrubSearchFilterProps> = ({
+  isModalVisible,
+  onClose,
+  activeFilters,
+  onActiveFilterChange,
+}) => {
+  // Individual filter states
+  const [activeHeightFilters, setActiveHeightFilters] = useState({
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:44]: Move height filters to a boolean in other filters as 'lowGrowing'

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+      Object.values(activeBloomFilters).some(Boolean) ||
+      Object.values(activeWaterFilters).some(Boolean) ||
+      Object.values(activeGrowthFilters).some(Boolean) ||
+      Object.values(activeOtherFilters).some(Boolean)
+    );
+  };
+
+  useEffect(() => {
+    onActiveFilterChange({
+      max_height: Object.keys(activeHeightFilters).filter(
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:92]: Remove max_height

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+  });
+
+  const [activeOtherFilters, setActiveOtherFilters] = useState({
+    californiaNative: activeFilters.other.includes('californiaNative'),
+    evergreen: activeFilters.other.includes('evergreen'),
+  });
+
+  const anyFiltersActive = () => {
+    return (
+      Object.values(activeHeightFilters).some(Boolean) ||
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:81]: Remove

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+    high: activeFilters.water_use.includes('high'),
+  });
+
+  const [activeGrowthFilters, setActiveGrowthFilters] = useState({
+    low: activeFilters.water_use.includes('slow'),
+    moderate: activeFilters.water_use.includes('moderate'),
+    high: activeFilters.water_use.includes('fast'),
+  });
+
+  const [activeOtherFilters, setActiveOtherFilters] = useState({
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:74]: Add height

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+      ) as string[],
+      growth_rate: Object.keys(activeGrowthFilters).filter(
+        key => activeGrowthFilters[key as keyof typeof activeGrowthFilters],
+      ) as string[],
+      other: Object.keys(activeOtherFilters).filter(
+        key => activeOtherFilters[key as keyof typeof activeOtherFilters],
+      ) as string[],
+    });
+  }, [
+    activeHeightFilters,
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:112]: Remove

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+    activeBloomFilters,
+    activeSunFilters,
+    activeWaterFilters,
+    activeOtherFilters,
+    onActiveFilterChange,
+  ]);
+
+  useEffect(() => {}, [
+    activeHeightFilters,
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:121]: Remove

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+      other: Object.keys(activeOtherFilters).filter(
+        key => activeOtherFilters[key as keyof typeof activeOtherFilters],
+      ) as string[],
+    });
+  }, [
+    activeHeightFilters,
+    activeBloomFilters,
+    activeSunFilters,
+    activeWaterFilters,
+    activeOtherFilters,
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:116]: Add growth

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+    onActiveFilterChange,
+  ]);
+
+  useEffect(() => {}, [
+    activeHeightFilters,
+    activeBloomFilters,
+    activeSunFilters,
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:123]: Add growth

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+    activeWaterFilters,
+    activeOtherFilters,
+  ]);
+
+  const toggleHeightFilter = (key: keyof typeof activeHeightFilters) => {
+    setActiveHeightFilters(prev => ({
+      ...prev,
+      [key]: !prev[key],
+    }));
+  };
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:133]: Remove

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+
+  const toggleWaterFilter = (key: keyof typeof activeWaterFilters) => {
+    setActiveWaterFilters(prev => ({
+      ...prev,
+      [key]: !prev[key],
+    }));
+  };
+
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:148]: Add filters for sun and growth

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 12]
```diff
@@ -0,0 +1,284 @@
...
+  const resetFilters = () => {
+    setActiveHeightFilters({
+      lowGrowing: false,
+      notLowGrowing: false,
+    });
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:160]: Remove

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 13]
```diff
@@ -0,0 +1,284 @@
...
+      winter: false,
+      spring: false,
+      summer: false,
+      fall: false,
+    });
+    setActiveWaterFilters({
+      low: false,
+      moderate: false,
+      high: false,
+    });
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:171]: Add filters for sun and growth

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 14]
```diff
@@ -0,0 +1,284 @@
...
+    setActiveOtherFilters({
+      californiaNative: false,
+      evergreen: false,
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:174]: Add height

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 15]
```diff
@@ -0,0 +1,284 @@
...
+    });
+    onActiveFilterChange({
+      max_height: [],
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:177]: Remove

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 16]
```diff
@@ -0,0 +1,284 @@
...
+            {/* Height */}
+            <View style={styles.filterProperties}>
+              <Text style={styles.subheaderText}>Height</Text>
+              <View style={styles.checkboxGroup}>
+                <Checkbox
+                  label="Low Growing (&lt; 2')"
+                  isChecked={activeHeightFilters.lowGrowing}
+                  onChange={() => toggleHeightFilter('lowGrowing')}
+                />
+                <Checkbox
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:228]: Remove

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 17]
```diff
@@ -0,0 +1,284 @@
...
+                  label="California native"
+                  isChecked={activeOtherFilters.californiaNative}
+                  onChange={() => toggleOtherFilter('californiaNative')}
+                />
+                <Checkbox
+                  label="Evergreen"
+                  isChecked={activeOtherFilters.evergreen}
+                  onChange={() => toggleOtherFilter('evergreen')}
+                />
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:271]: Add height

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx [hunk 18]
```diff
@@ -0,0 +1,284 @@
...
+                  onChange={() => toggleWaterFilter('moderate')}
+                />
+                <Checkbox
+                  label="High"
+                  isChecked={activeWaterFilters.high}
+                  onChange={() => toggleWaterFilter('high')}
+                />
+              </View>
+            </View>
+
```
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:257]: Add filters for bloom, sun, and growth

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx
Comment (carolynzhuang) [inline src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:1]: Remove evergreen, i realized they dont have that property for shrubs

File: src/screens/ShrubSearch/ShrubSpeciesSearch.tsx [hunk 19]
```diff
@@ -0,0 +1,218 @@
...
+type ShrubSpeciesSearchScreenProps = NativeStackScreenProps<
+  HomeStackParamList,
+  'ShrubSpeciesSearch'
+>;
+
+type shrubSpeciesCard = {
+  name: string;
+  imageURL: string;
+  availableStock: number;
+  isEvergreen: boolean;
```
Comment (carolynzhuang) [inline src/screens/ShrubSearch/ShrubSpeciesSearch.tsx:31]: Change to string called Dormancy

File: src/screens/ShrubSearch/ShrubSpeciesSearch.tsx [hunk 20]
```diff
@@ -0,0 +1,218 @@
...
+  sunExposure: string;
+  soilNeeds: string;
+  growthRate: string;
+  waterUse: string;
+  isCaliforniaNative: boolean;
+  isLowGrowing: boolean;
+};
+
+type ActiveFilters = {
+  max_height: string[];
```
Comment (carolynzhuang) [inline src/screens/ShrubSearch/ShrubSpeciesSearch.tsx:43]: Remove

File: src/screens/ShrubSearch/ShrubSpeciesSearch.tsx [hunk 21]
```diff
@@ -0,0 +1,218 @@
...
+
+export const ShrubSpeciesSearchScreen: React.FC<
+  ShrubSpeciesSearchScreenProps
+> = ({ navigation }) => {
+  const [shrubSpeciesCards, setShrubSpeciesCards] = useState<
+    shrubSpeciesCard[]
+  >([]);
+  const [searchText, setSearchText] = useState<string>('');
+  const [activeFilters, setActiveFilters] = useState<ActiveFilters>({
+    max_height: [],
```
Comment (carolynzhuang) [inline src/screens/ShrubSearch/ShrubSpeciesSearch.tsx:59]: Remove

File: src/screens/ShrubSearch/ShrubSpeciesSearch.tsx [hunk 22]
```diff
@@ -0,0 +1,218 @@
...
+          }),
+        );
+        setShrubSpeciesCards(cards);
+      }
+    };
+    loadShrubSpeciesData();
+  }, [isUserAdmin]);
+
+  const applyFilters = (shrub: shrubSpeciesCard) => {
+    if (activeFilters.max_height.length > 0) {
```
Comment (carolynzhuang) [inline src/screens/ShrubSearch/ShrubSpeciesSearch.tsx:114]: Move to other

File: src/screens/TreeSpeciesSearch/styles.ts [hunk 23]
```diff
@@ -68,4 +68,9 @@ export const styles = StyleSheet.create({
     ...typography.mediumRegular,
     color: colors.gray2,
   },
+
+  treeShrubToggle: {
```
Comment (carolynzhuang) [inline src/screens/TreeSpeciesSearch/styles.ts:72]: Remove, we are not using toggle anymore

File: src/types/navigation.ts [hunk 24]
```diff
@@ -5,10 +5,16 @@ export type LoginStackParamList = {
...
 
 export type HomeStackParamList = {
+  Home: undefined;
   TreeSpeciesSearch: undefined;
+  ShrubSpeciesSearch: undefined;
   QRCodeScanner: undefined;
   TreeInfo: { treeId: string };
   TreeSpeciesInfo: { speciesName: string };
+  ShrubSpeciesInfo: { speciesName: string };
+  ShrubSearchFilter: undefined;
```
Comment (carolynzhuang) [inline src/types/navigation.ts:15]: Remove filter from navigation

File: src/screens/ShrubSearch/ShrubSpeciesSearch.tsx [hunk 25]
```diff
@@ -0,0 +1,218 @@
...
+      const height = Number(shrub.dimension.split('x', 1)[0]);
```
Comment (carolynzhuang) [inline src/screens/ShrubSearch/ShrubSpeciesSearch.tsx:115]: change so that it can split on both lowercase and uppercase x

File: src/screens/ShrubSearch/ShrubSpeciesSearch.tsx
Comment (carolynzhuang) [inline src/screens/ShrubSearch/ShrubSpeciesSearch.tsx:1]: Rename folder to ShrubSpeciesSearch for consistency

**Review by carolynzhuang — COMMENTED** [review]
yayyy looks good! make sure to implement all of the filters, added some comments on small changes
I think adding the filters will fix the lint issues

File: src/screens/ShrubSearch/ShrubSpeciesSearch.tsx (hunk 25 above)
Comment (carolynzhuang) [reply src/screens/ShrubSearch/ShrubSpeciesSearch.tsx:115]: also does not split if its undefined

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx (hunk 17 above)
Comment (plaanty) [reply src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:271]: wait what do you mean "add height"? like a checkbox for "low growing" or something else?

File: src/components/ShrubSearchFilter/ShrubSearchFilter.tsx (hunk 5 above)
Comment (plaanty) [reply src/components/ShrubSearchFilter/ShrubSearchFilter.tsx:74]: what do you mean by "add height"? I changed "high" to "fast" as an edit to better reflect the enumerated types for activeGrowthFilters, but other than that I'm not sure what you mean
